server/app/services/export/QuestionJsonPresenter.java

```diff
@@ -0,0 +1,275 @@
+          question
+              .getApplicantQuestion()
+              .createFileUploadQuestion()
+              .getFileKeyValue()
+              .map(
+                  fileKey ->
+                      baseUrl
+                          + controllers.routes.FileController.acledAdminShow(
+                                  URLEncoder.encode(fileKey, StandardCharsets.UTF_8))
+                              .url())
+              .orElse(""));
+    }
+  }
+
+  class MultiOptionJsonPresenter implements QuestionJsonPresenter<MultiSelectQuestion> {
+
+    @Override
+    public ImmutableMap<Path, ImmutableList<String>> getJsonEntries(MultiSelectQuestion question) {
+      Path path = question.getSelectionPath().asApplicationPath();
+
+      if (question.getSelectedOptionsValue().isPresent()) {
+        ImmutableList<String> selectedOptions =
+            question.getSelectedOptionsValue().get().stream()
+                .map(LocalizedQuestionOption::optionText)
+                .collect(ImmutableList.toImmutableList());
+
+        return ImmutableMap.of(path, selectedOptions);
+      }
+
+      return ImmutableMap.of();
+    }
+  }
+
+  class ContextualizedScalarsJsonPresenter implements QuestionJsonPresenter<Question> {
+    @Override
+    public ImmutableMap<Path, String> getJsonEntries(Question question) {
+      return question.getApplicantQuestion().getContextualizedScalars().keySet().stream()
+          .filter(path -> !Scalar.getMetadataScalarKeys().contains(path.keyName()))
+          .collect(
+              toImmutableMap(
+                  path -> path,
+                  path ->
+                      question
+                          .getApplicantQuestion()
+                          .getApplicantData()
+                          .readAsString(path)
+                          .orElse("")));
+    }
+  }
+
+  class CurrencyJsonPresenter implements QuestionJsonPresenter<CurrencyQuestion> {
+    @Override
+    public ImmutableMap<Path, Double> getJsonEntries(CurrencyQuestion question) {
+      Path path =
+          question.getCurrencyPath().asApplicationPath().replacingLastSegment("currency_dollars");
+
+      if (question.getCurrencyValue().isPresent()) {
+        Long centsTotal = Long.valueOf(question.getCurrencyValue().get().getCents());
+
+        return ImmutableMap.of(path, centsTotal.doubleValue() / 100.0);
+      } else {
+        return ImmutableMap.of();
+      }
+    }
+  }
+
+  class DateJsonPresenter implements QuestionJsonPresenter<DateQuestion> {
+    @Override
+    public ImmutableMap<Path, String> getJsonEntries(DateQuestion question) {
+      Path path = question.getDatePath().asApplicationPath();
+
+      if (question.getDateValue().isPresent()) {
+        LocalDate date = question.getDateValue().get();
+        return ImmutableMap.of(path, DateTimeFormatter.ISO_DATE.format(date));
+      } else {
+        return ImmutableMap.of();
+      }
+    }
+  }
```

Comment:
Above code is taken from `case DATE`:

https://github.com/civiform/civiform/blob/358ad3dbb6e67aacb91afb3ede60e32ac281ba26/server/app/services/export/JsonExporter.java#L121-L219